compare_template_text/normalize_template_text.py

```diff
@@ -9,90 +9,71 @@ class NormalizeTemplate:
     def __init__(self,text_string,template_string):
         self.text_string = text_string
         self.template_string = template_string
+        self.template_string = self.template_string.replace('\\','@')
+        self.template_string  = re.escape(self.template_string)
                 
     def normalize_template(self):
-        self.remove_bulletted_text()
+        self.remove_repeating_chars()
         self.remove_replaceable_text()
         self.remove_omitable_text()
     
     def return_normalized_template(self):
         
         """ Returns the Normalized template after normalization techniques. """
-        print(self.template_string)
+        # print(self.template_string)
         return self.template_string
     
     def return_normalized_text(self):
         """ Returns the normalized text after normalization techniques. """
         
-        print(self.text_string)
         return self.text_string
     
     def remove_omitable_text(self):
         """ Removes text from the template and the text that is optional. So, the text 
         which is optional and present matches and gets replaced by '' while the text that doesn't match will 
         remain as it is thus giving an unmatch. """
          
-        lists = re.findall('(?<=\<\<beginoptional\>\>).*?(?=\<\<endoptional\>\>)',
+        lists = re.findall('`?<<beginoptional>>.*?<<endoptional>>`?',
                            self.template_string)
-        # print(lists)
+        
         for x in lists:
-            if(x.startswith('*')):
-                x = x[1:]
-
-            if(x.endswith('*')):
-                x = x[:-1]
-
-            self.text_string = self.text_string.replace(x,'')
-            self.template_string = self.template_string.replace(x,'')
-            self.template_string = self.template_string.replace('<<beginoptional>>','')
-            self.template_string = self.template_string.replace('<<endoptional>>','')
-            self.remove_repeating_chars()
+            match_regex = re.search('(?<=<<beginoptional>>).*?(?=<<endoptional>>)',x).group(0)
+            if(match_regex.startswith('`')):
+                match_regex = match_regex[1:]
+
+            if(match_regex.endswith('`')):
+                match_regex = match_regex[:-1]
```

Comment:
@rtgdk we cant entirely strip the "`" character since it's possible it is used in the middle of the match_regex. We just need to remove it from the beginning and the end.

Reply:
@anshuldutt21 So is it possible that match_regex can be equal to ``` ``text``  ``` ? Because strip will only remove it from the end and beginning. And if the text is like this - ``` ``text``  ``` - shouldn't we remove all the leading and trailing `` ` `` character?

Reply:
@rtgdk ` ``text`` ` is not possible since I am removing all the repeating \`  in the beginning of the file in the function remove_repeating_chars so \` will come only once in the beginning and in the end. It can come in the middle but we can't remove the middle \`.

Reply:
@anshuldutt21 strip won't remove it from the middle, it only trims from beginning and end.

Reply:
@anshuldutt21 strip will work here.